### PR TITLE
Add documentation for CA1870

### PR DESCRIPTION
**Pull request by MihaZupan**
Adds the code analyzer/fixer docs for https://github.com/dotnet/runtime/issues/78587
Followup after https://github.com/dotnet/roslyn-analyzers/pull/6898

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/code-analysis/quality-rules/ca1870.md](https://github.com/dotnet/docs/blob/a16d883d6a1be1e4404b78dbda11dc6f2288881c/docs/fundamentals/code-analysis/quality-rules/ca1870.md) | [CA1870: Use a cached 'SearchValues' instance](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1870?branch=pr-en-us-37056) |
| [docs/fundamentals/code-analysis/quality-rules/index.md](https://github.com/dotnet/docs/blob/a16d883d6a1be1e4404b78dbda11dc6f2288881c/docs/fundamentals/code-analysis/quality-rules/index.md) | [Code quality rules overview](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/index?branch=pr-en-us-37056) |
| [docs/fundamentals/code-analysis/quality-rules/performance-warnings.md](https://github.com/dotnet/docs/blob/a16d883d6a1be1e4404b78dbda11dc6f2288881c/docs/fundamentals/code-analysis/quality-rules/performance-warnings.md) | [Performance rules](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/performance-warnings?branch=pr-en-us-37056) |

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca1870.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+# CA1870: Use a cached 'SearchValues' instance
+
+| Property                            | Value                                  |
+|-------------------------------------|----------------------------------------|
+| **Rule ID**                         | CA1870                                 |
+| **Title**                           | Use a cached 'SearchValues' instance   |
+| **Category**                        | [Performance](performance-warnings.md) |
+| **Fix is breaking or non-breaking** | Non-breaking                           |
+| **Enabled by default in .NET 7**    | No                                     |
```
Comment (MihaZupan) [inline docs/fundamentals/code-analysis/quality-rules/ca1870.md:25]: I copied this from existing CA docs, does it need changing?

This analyzer is being added in .NET 8, where it should be enabled by default.

File: docs/fundamentals/code-analysis/quality-rules/performance-warnings.md [hunk 2]
```diff
@@ -74,3 +74,4 @@ Performance rules support high-performance libraries and applications.
...
 | [CA1869: Cache and reuse 'JsonSerializerOptions' instances](ca1869.md) | Using a local instance of <xref:System.Text.Json.JsonSerializerOptions> for serialization or deserialization can substantially degrade the performance of your application if your code executes multiple times, since System.Text.Json internally caches serialization-related metadata into the provided instance. |
+| [CA1870: Use a cached 'SearchValues' instance](ca1870.md) | Using a cached  <xref:System.Buffers.SearchValues%601> instance is more efficient than passing values to 'IndexOfAny'/'ContainsAny' directly. |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/performance-warnings.md:77]:
```suggestion
| [CA1870: Use a cached 'SearchValues' instance](ca1870.md) | Using a cached  <xref:System.Buffers.SearchValues%601> instance is more efficient than passing values to 'IndexOfAny' or 'ContainsAny' directly. |
```

File: docs/fundamentals/code-analysis/quality-rules/index.md [hunk 3]
```diff
@@ -161,6 +161,7 @@ The following table lists code quality analysis rules.
...
 > | [CA1869: Cache and reuse 'JsonSerializerOptions' instances](ca1869.md) | Using a local instance of <xref:System.Text.Json.JsonSerializerOptions> for serialization or deserialization can substantially degrade the performance of your application if your code executes multiple times, since System.Text.Json internally caches serialization-related metadata into the provided instance. |
+> | [CA1870: Use a cached 'SearchValues' instance](ca1870.md) | Using a cached <xref:System.Buffers.SearchValues%601> instance is more efficient than passing values to 'IndexOfAny'/'ContainsAny' directly. |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/index.md:164]:
```suggestion
> | [CA1870: Use a cached 'SearchValues' instance](ca1870.md) | Using a cached <xref:System.Buffers.SearchValues%601> instance is more efficient than passing values to 'IndexOfAny' or 'ContainsAny' directly. |
```

File: docs/fundamentals/code-analysis/quality-rules/ca1870.md (hunk 1 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/quality-rules/ca1870.md:25]: Yes, let's update it to .NET 8.

File: docs/fundamentals/code-analysis/quality-rules/ca1870.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+## Cause
+
+An `IndexOfAny`/`ContainsAny` method is being called with many constant values in a way that can benefit from using <xref:System.Buffers.SearchValues> instead.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1870.md:29]:
```suggestion
An `IndexOfAny` or `ContainsAny` method is called with many constant values in a way that can benefit from using <xref:System.Buffers.SearchValues> instead.
```

File: docs/fundamentals/code-analysis/quality-rules/ca1870.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+The rule will not flag calls using up to 5 values, as those are already using an optimal implementation.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1870.md:31]:
```suggestion
The rule doesn't flag calls that use up to five values, as those already use an optimal implementation.
```

File: docs/fundamentals/code-analysis/quality-rules/ca1870.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+## Rule description
+
+Using a cached <xref:System.Buffers.SearchValues%601> instance is more efficient than passing values to `IndexOfAny`/`ContainsAny` directly.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1870.md:35]:
```suggestion
Using a cached <xref:System.Buffers.SearchValues%601> instance is more efficient than passing values to `IndexOfAny` or `ContainsAny` directly.
```

File: docs/fundamentals/code-analysis/quality-rules/ca1870.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+## How to fix violations
+
+Create and cache a <xref:System.Buffers.SearchValues%601> instance in a `static readonly` field, then pass that instance to the `IndexOfAny`/`ContainsAny` call instead.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1870.md:39]:
```suggestion
Create and cache a <xref:System.Buffers.SearchValues%601> instance in a `static readonly` field, then pass that instance to the `IndexOfAny` or `ContainsAny` call instead.
```

**Review by gewarren — APPROVED** [review]
Can you also add the new file to the TOC at https://github.com/dotnet/docs/blob/5ddf14ea434f6887b633af829439345d76ba366c/docs/navigate/tools-diagnostics/toc.yml#L714? Thanks for adding this documentation!

**Pull request merged**